server/main_server/src/agent_main.cpp

```diff
@@ -0,0 +1,1326 @@
+            const auto& config{config_handle.get_json()};
+
+            for (const auto& item : config.at(irods::KW_CFG_PLUGIN_CONFIGURATION).items()) {
+                for (const auto& plugin : item.value().items()) {
+                    deinit_shared_memory_for_plugin(plugin.value());
+                }
+            }
+        }
+        catch (const irods::exception& e) {
+            return irods::error(e);
+        }
+        catch (const std::exception& e) {
+            return ERROR(SYS_INTERNAL_ERR, e.what());
+        }
+
+        return SUCCESS();
+    } // deinit_shared_memory_for_plugins
+
+    // TODO Remove this since initAgent() covers this for agents.
+#if 0
+    auto initServer(RsComm& _comm) -> int
+    {
+        if (const auto ec = initServerInfo(0, &_comm); ec < 0) {
+            log_af::info("{}: initServerInfo error, status = {}", __func__, ec);
+            return ec;
+        }
+
+        // TODO Unnecessary?
+        //resc_mgr.print_local_resources();
+
+        printZoneInfo();
+
+        rodsServerHost_t* rodsServerHost{};
+        if (const auto ec = getRcatHost(PRIMARY_RCAT, nullptr, &rodsServerHost); ec < 0 || !rodsServerHost) {
+            return ec;
+        }
+
+        std::string svc_role;
+        if (const auto res = get_catalog_service_role(svc_role); !res.ok()) {
+            log_af::error("{}: Could not get server role. [error code={}]", __func__, res.code());
+            return res.code();
+        }
+
+        if (LOCAL_HOST == rodsServerHost->localFlag) {
+            if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+                disconnectRcat();
+            }
+        }
+        else if (rodsServerHost->conn) {
+            rcDisconnect(rodsServerHost->conn);
+            rodsServerHost->conn = nullptr;
+        }
+
+        return 0;
+    } // initServer
+#endif
+
+    auto initServerMain(RsComm& _comm) -> int
+    {
+        std::memset(&_comm, 0, sizeof(RsComm));
+        int status = getRodsEnv(&_comm.myEnv);
+        if (status < 0) {
+            log_af::error("{}: getRodsEnv error. status = {}", __func__, status);
+            return status;
+        }
+
+        // TODO Verify this is reading from the correct location.
+        // Consider non-pkg installs and pkg installs.
+        setRsCommFromRodsEnv(&_comm);
+
+        // TODO Verify this is okay to do in the agent factory before forking any agents.
+        // Need a script that stress tests the server.
+#if 0
+        // Load server API table so that API plugins which are needed to stand up the server are
+        // available for use.
+        irods::api_entry_table& RsApiTable = irods::get_server_api_table();
+        irods::pack_entry_table& ApiPackTable = irods::get_pack_table();
+        if (const auto err = irods::init_api_table(RsApiTable, ApiPackTable, false); !err.ok()) {
+            irods::log(PASS(err));
+            return err.code();
+        }
```

Comment:
This code is from 4.3 and is unchanged.

@alanking Do we have an issue number for this? If so, then I'll attach it to the TODO comment.

Reply:
For which part? The script for stress testing the server, or verifying that it is okay load the server API table in the agent factory before forking agents? Neither sounds familiar, but just wanted to confirm.

Reply:
Sorry, this is the wrong code block. There's another in here for the auth plugins. Let me check on that.

Reply:
NTS: Remove this code block as the server seems to behave as intended during testing.

Reply:
Done.